src/main/java/org/jenkinsci/remoting/protocol/IOHub.java

```diff
@@ -433,11 +450,18 @@ public final void run() {
                         // in an immediately ready selection key, hence we use the non-blocking form
                         selected = selector.selectNow();
                     } else {
-                        // We just wait for one second if the data is not available
-                        // Before Remoting 3.15 it was an infinite wait, which was causing hanging of the test logic
-                        // If the timeout happens, we will get 0 in selected, and all statistics calculation below will be skipped.
-                        selected = selector.select(1000);
```

Comment:
so another alternative which seems to me to be slightly better is to just use 
`selected = selector.select();`

and then create a thread that every one second does selector.wakeup() because we already have to handle spurious wakeups and then you avoid keeping lots of state with the timeout and `selectorPollTimestamp` not being thread safe because its not atomic being 64bit and not volatile...

Reply:
As I said, the impl is lame :) 
Just wanted to dry run the concept